File: incubator/net.sf.j2s.test.mauve/j2slib/org/eclipse/core/runtime/internal/stats/BundleStats.js
Clazz.declarePackage ("org.eclipse.core.runtime.internal.stats");
Clazz.load (["java.util.ArrayList"], "org.eclipse.core.runtime.internal.stats.BundleStats", ["org.eclipse.core.runtime.internal.stats.ClassloaderStats", "$.StatsManager"], function () {
c$ = Clazz.decorateAsClass (function () {
this.symbolicName = null;
this.id = 0;
this.activationOrder = 0;
this.timestamp = 0;
this.duringStartup = false;
this.startupTime = 0;
this.startupMethodTime = 0;
this.traceStart = -1;
this.traceEnd = -1;
this.bundlesActivated = null;
this.activatedBy = null;
Clazz.instantialize (this, arguments);
}, org.eclipse.core.runtime.internal.stats, "BundleStats");
Clazz.prepareFields (c$, function () {
this.bundlesActivated =  new java.util.ArrayList (3);
});
Clazz.makeConstructor (c$, 
function (name, id) {
this.symbolicName = name;
this.id = id;
}, "~S,~N");
Clazz.defineMethod (c$, "getTimestamp", 
function () {
return this.timestamp;
});
Clazz.defineMethod (c$, "getActivationOrder", 
function () {
return this.activationOrder;
});
Clazz.defineMethod (c$, "activated", 
function (info) {
this.bundlesActivated.add (info);
}, "org.eclipse.core.runtime.internal.stats.BundleStats");
Clazz.defineMethod (c$, "getActivatedBy", 
function () {
return this.activatedBy;
});
Clazz.defineMethod (c$, "getId", 
function () {
return this.id;
});
Clazz.defineMethod (c$, "getSymbolicName", 
function () {
return this.symbolicName;
});
Clazz.defineMethod (c$, "getStartupTime", 
function () {
return this.startupTime;
});
Clazz.defineMethod (c$, "getStartupMethodTime", 
function () {
return this.startupMethodTime;
});
Clazz.defineMethod (c$, "isStartupBundle", 
function () {
return this.duringStartup;
});
Clazz.defineMethod (c$, "getClassLoadCount", 
function () {
if (!org.eclipse.core.runtime.internal.stats.StatsManager.MONITOR_CLASSES) return 0;
var loader = org.eclipse.core.runtime.internal.stats.ClassloaderStats.getLoader (this.symbolicName);
return loader == null ? 0 : loader.getClassLoadCount ();
});
Clazz.defineMethod (c$, "getClassLoadTime", 
function () {
if (!org.eclipse.core.runtime.internal.stats.StatsManager.MONITOR_CLASSES) return 0;
var loader = org.eclipse.core.runtime.internal.stats.ClassloaderStats.getLoader (this.symbolicName);
return loader == null ? 0 : loader.getClassLoadTime ();
});
Clazz.defineMethod (c$, "getBundlesActivated", 
function () {
return this.bundlesActivated;
});
Clazz.defineMethod (c$, "getTraceStart", 
function () {
return this.traceStart;
});
Clazz.defineMethod (c$, "getTraceEnd", 
function () {
return this.traceEnd;
});
Clazz.defineMethod (c$, "setTimestamp", 
function (value) {
this.timestamp = value;
}, "~N");
Clazz.defineMethod (c$, "setActivationOrder", 
function (value) {
this.activationOrder = value;
}, "~N");
Clazz.defineMethod (c$, "setTraceStart", 
function (time) {
this.traceStart = time;
}, "~N");
Clazz.defineMethod (c$, "setDuringStartup", 
function (value) {
this.duringStartup = value;
}, "~B");
Clazz.defineMethod (c$, "endActivation", 
function () {
this.startupTime = System.currentTimeMillis () - this.timestamp;
});
Clazz.defineMethod (c$, "setTraceEnd", 
function (position) {
this.traceEnd = position;
}, "~N");
Clazz.defineMethod (c$, "setActivatedBy", 
function (value) {
this.activatedBy = value;
}, "org.eclipse.core.runtime.internal.stats.BundleStats");
});

File: old/sources/net.sf.j2s.java.org.eclipse.osgi/src/org/eclipse/core/runtime/internal/stats/BundleStats.java
/*******************************************************************************
 * Copyright (c) 2000, 2004 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.eclipse.core.runtime.internal.stats;

import java.util.ArrayList;

/**
 * Contains information about activated bundles and acts as the main 
 * entry point for logging bundle activity.
 */

public class BundleStats {
	public String symbolicName;
	public long id;
	public int activationOrder;
	private long timestamp; //timeStamp at which this bundle has been activated
	private boolean duringStartup; // indicate if the bundle has been activated during startup
	private long startupTime; // the time took by the bundle to startup
	private long startupMethodTime; // the time took to run the startup method

	// Indicate the position of the activation trace in the file
	private long traceStart = -1;
	private long traceEnd = -1;

	//To keep bundle parentage
	private ArrayList bundlesActivated = new ArrayList(3); // TODO create lazily
	private BundleStats activatedBy = null;

	public BundleStats(String name, long id) {
		this.symbolicName = name;
		this.id = id;
	}

	public long getTimestamp() {
		return timestamp;
	}

	public int getActivationOrder() {
		return activationOrder;
	}

	protected void activated(BundleStats info) {
		bundlesActivated.add(info);
	}

	public BundleStats getActivatedBy() {
		return activatedBy;
	}

	public long getId() {
		return id;
	}

	public String getSymbolicName() {
		return symbolicName;
	}

	public long getStartupTime() {
		return startupTime;
	}

	public long getStartupMethodTime() {
		return startupMethodTime;
	}

	public boolean isStartupBundle() {
		return duringStartup;
	}

	public int getClassLoadCount() {
		if (!StatsManager.MONITOR_CLASSES)
			return 0;
		ClassloaderStats loader = ClassloaderStats.getLoader(symbolicName);
		return loader == null ? 0 : loader.getClassLoadCount();
	}

	public long getClassLoadTime() {
		if (!StatsManager.MONITOR_CLASSES)
			return 0;
		ClassloaderStats loader = ClassloaderStats.getLoader(symbolicName);
		return loader == null ? 0 : loader.getClassLoadTime();
	}

	public ArrayList getBundlesActivated() {
		return bundlesActivated;
	}

	public long getTraceStart() {
		return traceStart;
	}

	public long getTraceEnd() {
		return traceEnd;
	}

	protected void setTimestamp(long value) {
		timestamp = value;
	}

	protected void setActivationOrder(int value) {
		activationOrder = value;
	}

	protected void setTraceStart(long time) {
		traceStart = time;
	}

	protected void setDuringStartup(boolean value) {
		duringStartup = value;
	}

	protected void endActivation() {
		startupTime = System.currentTimeMillis() - timestamp;
	}

	protected void setTraceEnd(long position) {
		traceEnd = position;
	}

	protected void setActivatedBy(BundleStats value) {
		activatedBy = value;
	}
}
